mr/readme.md

```diff
@@ -110,8 +112,8 @@ Some forms of IO storage (for example, Google Storage) work better when you read
 
 Note that the idealized model of a thread per CPU doesn't actually work on Linux when reading files from local disk. Linux doesn't support async IO for any filesystem that is not XFS. Because of this, there is a separate pool of threads that only run IO calls and send their results to the caller. This problem doesn't exist when working with Google Storage, since Asio is capable of handling asynchronous network IO.
 
-TODO: Write about runner.
+Although both types of executors handle the logic of the mapping/joining process, they try to avoid doing the I/O operation directly. Instead, I/O operations are implemented by a `Runner` object which represents the interactions of the MR infrastructure with the disk/net.
```

Comment:
they avoid interacting with the external environment directly. Instead, all I/O operations  are abstracted away using a `Runner` object which represents the external environment of the MR infrastructure.